bopscrk/modules/auxiliars.py

```diff
@@ -49,13 +46,12 @@ def append_wordlist_to_file(filepath, wordlist):
 
 def remove_duplicates_from_file(infile_path, outfile_path="temp.000000000.bopscrk"):
     lines_seen = set()  # holds lines already seen
-    outfile = open(outfile_path, "w")
-    infile = open(infile_path, "r")
-    for line in infile:
-        if line not in lines_seen:  # not a duplicate
-            outfile.write(line)
-            lines_seen.add(line)
-    outfile.close()
+    with open(outfile_path, "w") as outfile:
+        infile = open(infile_path, "r")
+        for line in infile:
+            if line not in lines_seen:  # not a duplicate
+                outfile.write(line)
+                lines_seen.add(line)
```

Comment:
I would recommend sorting the data read here from `infile_path` before executing the loop, this will remove the need for a set(), reducing the memory footprint of this function.  The conditional check for a duplicate is a check if two strings are equal with only accessing those memory locations for comparison, where a `set` has to verify all values in the set that is does not exist (Many loops even if they are C level loops!).

I use `sorted` here but if the file is already sorted that can be removed.  

```python
def remove_duplicates_from_file(infile_path, outfile_path="temp.000000000.bopscrk"):
    last_line = '' # holds lines already seen
    with open(outfile_path, "w") as outfile:
        infile = open(infile_path, "r")
        for line in sorted(infile):
            if line != last_line:  # not a duplicate
                outfile.write(line)
                last_line = line  
```